dataland-frontend/tests/e2e/fixtures/eutaxonomy/non-financials/EuTaxonomyDataForNonFinancialsFixtures.ts

```diff
@@ -7,192 +7,114 @@ import {
 
 /**
- * Generates a random amount of money
- * @param undefinedProbabilityOfFields the probability of an undefined value per field
- * @returns an amount of money
+ * Generates a single fixture for the eutaxonomy-non-financials framework
+ * @param undefinedProbability the probability (as number between 0 and 1) for "undefined" values in nullable fields
+ * @returns the generated fixture
  */
-export function generateAmountWithCurrency(undefinedProbabilityOfFields?: number): AmountWithCurrency {
+export function generateEuTaxonomyDataForNonFinancials(
+  undefinedProbability = DEFAULT_PROBABILITY,
+): EuTaxonomyDataForNonFinancials {
+  const dataGenerator = new EuNonFinancialsGenerator(undefinedProbability);
   return {
-    amount: valueOrUndefined(randomFloat(1000000, 10000000000, 1), undefinedProbabilityOfFields),
-    currency: valueOrUndefined(generateIso4217CurrencyCode(), undefinedProbabilityOfFields),
+    general: generateEuTaxonomyWithBaseFields(dataGenerator.reports, undefinedProbability),
+    opex: dataGenerator.generateEuTaxonomyPerCashflowType(),
+    capex: dataGenerator.generateEuTaxonomyPerCashflowType(),
+    revenue: dataGenerator.generateEuTaxonomyPerCashflowType(),
   };
 }
 
-/**
- * Generates a random financial share
- * @param undefinedProbabilityOfFields the probability of an undefined value per field
- * @returns a financial share
- */
-export function generateFinancialShare(undefinedProbabilityOfFields?: number): RelativeAndAbsoluteFinancialShare {
-  return {
-    relativeShareInPercent: valueOrUndefined(randomPercentageValue(), undefinedProbabilityOfFields),
-    absoluteShare: valueOrUndefined(
-      generateAmountWithCurrency(undefinedProbabilityOfFields),
-      undefinedProbabilityOfFields,
-    ),
-  };
-}
+export class EuNonFinancialsGenerator extends Generator {
+  /**
+   * Generates a random amount of money
+   * @returns an amount of money
+   */
+  generateAmountWithCurrency(): AmountWithCurrency {
+    return {
+      amount: this.randomCurrencyValue(),
+      currency: this.valueOrUndefined(generateIso4217CurrencyCode()),
+    };
+  }
 
-/**
- * Generates a random activity
- * @param undefinedProbabilityOfFields the probability of an undefined value per field
- * @returns a random activity
- */
-function generateActivity(undefinedProbabilityOfFields?: number): EuTaxonomyActivity {
-  const randomActivityName: Activity = faker.helpers.arrayElement(Object.values(Activity));
-  return {
-    activityName: randomActivityName,
-    naceCodes: valueOrUndefined(
-      getRandomNumberOfNaceCodesForSpecificActivity(randomActivityName),
-      undefinedProbabilityOfFields,
-    ),
-    share: valueOrUndefined(generateFinancialShare(undefinedProbabilityOfFields), undefinedProbabilityOfFields),
-  };
-}
+  /**
+   * Generates a random financial share
+   * @returns a financial share
+   */
+  generateFinancialShare(): RelativeAndAbsoluteFinancialShare | undefined {
```

Comment:
By our naming convention. this should not be named "generate..." and then be able to return undefined, right?

Reply:
true, name was kept from old implementation and has been changed